src/core/stac/qgsstacasset.cpp

```diff
@@ -87,6 +96,8 @@ QgsMimeDataUtils::Uri QgsStacAsset::uri() const
          href().startsWith( QLatin1String( "ftp" ), Qt::CaseInsensitive ) )
     {
       uri.uri = QStringLiteral( "/vsicurl/%1" ).arg( href() );
+      if ( !authcfg.isEmpty() )
```

Comment:
Ensure description highlights that this logic was moved and consolidated, not added